jax/_src/cudnn/fused_attention_stablehlo.py

```diff
@@ -178,46 +191,56 @@ def create_dot_product_attention_backend_config(batch,
 def get_custom_call_name(has_bias, has_mask, has_dropout, is_bwd):
   return _custom_name_maps[(is_bwd, has_dropout, has_mask, has_bias)]
 
-def check_qkv_layout(query, key, value):
-  assert len(query.shape) == len(key.shape) == len(value.shape) == 4, \
-    "query, key and value should have rank 4."
-
-  # Only support fp16 and bf16 here
-  query_dtype = query.dtype
-  key_dtype = key.dtype
-  value_dtype = value.dtype
-  assert query_dtype == key_dtype == value_dtype and query_dtype in [jnp.float16, jnp.bfloat16], \
-    "query, key and value should have same dtype and should be float16 or bfloat16"
-
-  q_batch, q_seq_len, q_num_heads, q_head_dim = query.shape
-  k_batch, k_seq_len, k_num_heads, k_head_dim = key.shape
-  v_batch, v_seq_len, v_num_heads, v_head_dim = value.shape
-  if not((q_batch == k_batch == v_batch)
-      and (k_seq_len == v_seq_len)
-      and (q_num_heads == k_num_heads == v_num_heads)
-      and (q_head_dim == k_head_dim == v_head_dim)):
-    raise ValueError(
-      "query should have layout [batch, q_seq, num_heads, head_dim], " \
-      "key and value should have layout [batch, kv_seq, num_heads, head_dim].")
-
-def check_is_flash_attention(query, key, cudnn_version, has_bias, is_training):
-  batch, q_seq_len, num_heads, head_dim = query.shape
-  _, kv_seq_len, _, _ = key.shape
-
-  # check if attention pattern is supported by flash attention or fused attention
-  if q_seq_len <= 512 and kv_seq_len <= 512 and head_dim == 64 \
-    and (not is_training or q_seq_len % 64 == 0 and kv_seq_len % 64 == 0):
+def check_qkv_layout(query, key, value, layout):
+  def assert_eq(a, b, c, msg):
+    assert a == b == c, msg + f' must be same: {a}, {b}, {c}'
```

Comment:
I can see that the old version also used `assert` for argument validation, but since you are changing the validation logic slightly, I would recommend using `raise` instead. For example:

```
if q_rank != 4:
  raise ValueError(f"Q must have a ran of 4, got {q_rank}")
```

Reply:
Done.